### DISCUSSION TITLE
Mix 2d and 3d transforms

### DISCUSSION THREAD

**Discussion by lilipj** [Project-MONAI/MONAI#6566] [Q&A] [answered]
Hello,

I have 3d data, and i want to perform 2d training. But I need one transform to be done in 3d.
Is it possible to do this ?
Thanks in advance.

**Comment by KumoLiu** [comment, marked as answer]
Hi @lilipj, I think most of MONAI's transforms can support both 2d and 3d data.
Thanks!

**Reply by lilipj** [reply]
Yes, but I would like to do all 2D-3D tranforms in the same Compose() like
Compose(load_2d_transform, transform3d, transform2d,…)

**Reply by KumoLiu** [reply]
I don't fully understand you, for the same data, let's say 3d, how can you do both 2d and 3d transform on it? Could you please give more details, or put a pseudo code？

**Reply by lilipj** [reply]
Sorry, my explanations are not very clear, I don't have code yet. 
In fact, my inputs are all the 2d slices coming from a 3d volume, so, the loader is 2d.
But what i'm trying to do is to concatenate all slices from a same patient, to reconstruct volume and perform a 3d normalization on the full volume, I don't want normalization slice by slice.
Then, split the normalized volume to re-create 2d slices and perform other transforms.
I know the best way would be to perform directly from 3d input, and include a SplitDim to continue in 2D. But maybe it's possible to concatenate slices in a monai transform or maybe not !!

**Reply by KumoLiu** [reply]
Hi @lilipj, if I understand correctly, I think you want to find the transform which can help you concat the 2d files, maybe you can use `ConcatItemsd`.
https://github.com/Project-MONAI/MONAI/blob/e5979946ec2a71d33a862a41491216e03fb02660/monai/transforms/utility/dictionary.py#LL1016C17-L1016C17

Thanks!

**Reply by lilipj** [reply]
Thanks, i'm going to check that.
